openedx/core/djangoapps/user_api/preferences/tests/test_api.py

```diff
@@ -181,6 +181,34 @@ def test_update_user_preferences(self):
             "new_value"
         )
 
+    def test_update_user_preferences_with_username(self):
```

Comment:
@nasthagiri update is only accessible to the user itself; If requesting_user and user/username are different; it will throw authorization error and not authorized cases are handled in `test_update_user_preferences_errors`
These two (`test_update_user_preferences_with_username ` & `test_update_user_preferences_with_user`) are for success scenarios